### ISSUE TITLE
Proper names and introductions

### ISSUE THREAD

**Issue by davedotluebke** [davedotluebke/old-skool-text-game#8]
Players and creatures (or maybe just NPCs?) can have proper names. For example the short description might be "a dark-eyed swarthy gnome" while the proper name might be "Cedric". If you "introduce" yourself you give your proper name ("The dark-eyed swarthy gnome introduces himself as Cedric"), and from then on you would see the proper name when the creature's short description is accessed ("Cedric leaves to the west", or "Cedric is here" at the end of a room description, for example).

**Comment by davedotluebke** [comment]
Implemented this in 40a3355. Strings passed to `emit()` now support 'name tags' which are replaced by either a proper name or a short description. Articles ('the', 'a', 'an') are handled correctly. From the comment on `Creature.perceive()`:
```
def perceive(self, message):
        '''Parse a string passed to `emit()` and customize it for this
        player. Searches the string for special tags (indicated with the '&'
        symbol) and replaces the substring following that tag (up to a 
        whitespace character) with a customized substring. Currently supports
        the following tags, in which <id> is the ID attribute of an object O:
            tag      description
            -------  --------------------------------------------------------
            &nd<id>: 'name-definite': replace with the proper name of O, if O
                     has been introduced to this player, else the short
                     description of O preceeded by the definite article 'the'
            &nD<id>: 'name-capitalized-definite': replace with the proper name
                     of O, if O has been introduced to this player, else the
                     short description of O proceeded by the capitalized 
                     definite article 'The'
            &ni<id>: 'name-indefinite': replace with O.proper_name if O has 
                     been introduced, else O.short_desc preceeded by the 
                     indefinite article ('a' or 'an')
            &nI<id>: 'name-indefinite-capitalized': replace with O.proper_name
                     if O has been introduced, else 'A' or 'An' + O.short_desc
            &nn<id>: 'name-no-article': replace with O.proper_name if O has 
                     been introduced, else O.short_desc with no article.
        '''
```

However, only a few of the messages a player sees have been updated to use the new system. Before closing this issue, we should go through all `emit()` calls to update if necessary. There may be a few `cons.write()` calls that need to be converted to `cons.user.perceive()` to use the new name tags as well.

**Comment by davedotluebke** [comment]
Finished updating all emit() calls to use new syntax. Closing.
